xzre.h

```diff
@@ -68,6 +70,75 @@ typedef Elf64_Xword Elf64_Relr;
 #define __must_be_array(a) BUILD_BUG_ON_ZERO(__same_type((a), &(a)[0]))
 #define ARRAY_SIZE(arr) (sizeof(arr) / sizeof((arr)[0]) + __must_be_array(arr))
 
+// copied from https://sourceware.org/git/?p=glibc.git;a=blob;f=include/link.h;h=bef2820b40cd553c77990dcda4f4ccf0203a9110;hb=f94f6d8a3572840d3ba42ab9ace3ea522c99c0c2#l360
```

Comment:
I am not sure what the rules are for copying code from glibc - but since this repo is under the same licence I think this is ok?

if its decided we shouldn't do this then we can make things opaque pointers but will make the disassembled code a bit harder to read without a copy of glibc's header files handy

Reply:
We use the GPL here, and those headers are LGPL. Therefore it's not an issue.

For what it's worth, the LGPL states:

> «A program that contains no derivative of any portion of the Library, but is designed to work with the Library by being compiled or linked with it, is called a "work that uses the Library". Such a work, in isolation, is not a derivative work of the Library, and therefore falls outside the scope of this License.» 

One could argue that these data structures are interfaces for glibc (the library) and using them doesn't constitute derivative work, but i'm not a lawyer.
Either way, it's good for me.